src/vsr/replica.zig

```diff
@@ -4296,8 +4170,15 @@ pub fn Replica(
             assert(message.header.command == .do_view_change);
             assert(message.header.view == self.view);
             assert(message.header.op == self.op);
-            assert(message.header.op == self.message_body_as_headers(message)[0].op);
-            assert(message.header.commit == self.commit_max);
+            assert(message.header.op == message_body_as_headers(message)[0].op);
+            // Each replica must advertise its own commit number, so that the primary can know which
+            // headers must be replaced in its own log. Otherwise, a gap in the log may prevent the
+            // primary from repairing its own log adequately, resulting in the log being forked if
+            // the primary also discards uncommitted operations.
+            // It is also safe not to use `commit_max` here because the new primary will assume that
+            // operations after the highest `commit_min` may yet have been committed before the old
+            // primary crashed. The new primary will use the NACK protocol to be sure of a discard.
```

Comment:
This is still true, since we have two ways of discarding: for a clear gap, or otherwise via the NACK protocol. We just no longer use the NACK protocol where there is already a clear gap.